### ISSUE TITLE
Distributed metrics support

### ISSUE THREAD

**Issue by schmidt-ai** [pyg-team/pytorch_geometric#8000]
### 🛠 Proposed Refactor

Refactor `sklearn.metrics` API callsites in training loops/forward passes to use e.g. `torchmetrics` instead. The torchmetrics API supports accurate distributed metric computation. For example, in GAE, sklearn is used to compute average precision and ROC AUC. In a distributed setting, these metrics will be computed independently on each rank and then averaged across nodes, which is not correct. If we could replace these with the respective torchmetrics APIs we would get correct behavior.

This may be a case isolated to GAE since I see usage of torchmetrics elsewhere in the codebase from a quick skim.

### Suggest a potential alternative/fix

Use torchmetrics as described above.

**Comment by rusty1s** [comment]
Thanks for the issue. I am personally not a super huge fan of making `torchmetrics` required to run the examples, but I agree that it is necessary to run PyG in distributed mode. If you would like, we could move the GAE example to a distributed example and include it as part of `examples/multi_gpu`. WDYT?

**Comment by rusty1s** [comment]
Can you clarify what you mean?
